$NetBSD: patch-ak,v 1.1 2006/01/26 15:16:16 rillig Exp $

Declarations inside of macros are evil. Even more if they are not preceded
by an additional opening brace.

--- src/mesa/swrast/s_triangle.c.orig	2006-01-23 21:27:28.000000000 +0100
+++ src/mesa/swrast/s_triangle.c	2006-01-23 21:27:06.000000000 +0100
@@ -163,6 +163,7 @@ _swrast_culltriangle( GLcontext *ctx,
    }
 
 #define RENDER_SPAN( span  )						\
+{									\
    GLuint i;								\
    span.intTex[0] -= FIXED_HALF; /* off-by-one error? */		\
    span.intTex[1] -= FIXED_HALF;					\
@@ -177,7 +178,8 @@ _swrast_culltriangle( GLcontext *ctx,
       span.intTex[0] += span.intTexStep[0];				\
       span.intTex[1] += span.intTexStep[1];				\
    }									\
-   rb->PutRowRGB(ctx, rb, span.end, span.x, span.y, span.array->rgb, NULL);
+   rb->PutRowRGB(ctx, rb, span.end, span.x, span.y, span.array->rgb, NULL); \
+}
 
 #include "s_tritemp.h"
 
